packages/opencensus-exporter-jaeger/src/jaeger.ts

```diff
@@ -0,0 +1,197 @@
+  private timeoutSet = false;
+
+
+  constructor(options: JaegerTraceExporterOptions) {
+    const pjson = require('../../package.json');
+    this.logger = options.logger || logger.logger('debug');
+    this.bufferTimeout = options.bufferTimeout;
+    this.sender = new UDPSender(options);
+    const tags = options.tags || [];
+    tags[JaegerTraceExporter.JAEGER_OPENCENSUS_EXPORTER_VERSION_TAG_KEY] =
+        `opencensus-exporter-jaeger-${pjson.version}`;
+    tags[JaegerTraceExporter.TRACER_HOSTNAME_TAG_KEY] = os.hostname();
+    tags[JaegerTraceExporter.PROCESS_IP] = Utils.ipToInt(Utils.myIp());
+
+    this.process = {
+      serviceName: options.serviceName,
+      tags: options.tags ? ThriftUtils.getThriftTags(tags) : [],
+    };
+    this.sender.setProcess(this.process);
+  }
+
+  /**
+   * Is called whenever a span is ended.
+   * @param root the ended span
+   */
+  onEndSpan(root: types.RootSpan) {
+    this.logger.debug('onEndSpan: adding rootSpan: %s', root.name);
+    this.addSpanToSenderBuffer(root)
+        .then(result => {
+          this.addToBuffer(root, result as number);
+          for (const span of root.spans) {
+            this.addSpanToSenderBuffer(span)
+                .then(result => {
+                  this.addToBuffer(span, result as number);
+                })
+                .catch(err => {
+                  return;
+                });
+          }
+        })
+        .catch(err => {
+          return;
+        });
+
+    // Set a buffer timeout
+    this.setBufferTimeout();
+  }
+
+  /** Not used for this exporter */
+  onStartSpan(root: types.RootSpan) {}
+
+  private addToBuffer(span: types.Span, numSpans: number) {
+    if (numSpans > 0) {
+      this.successBuffer = this.successBuffer.concat(this.spanBuffer);
+      if (numSpans === this.spanBuffer.length) {
+        this.spanBuffer = [span];
+      } else {
+        this.spanBuffer = [];
+      }
+    } else {
+      this.logger.debug('adding to buffer %s', span.name);
+      this.spanBuffer.push(span);
+    }
+  }
+
+  private addSpanToSenderBuffer(span: types.Span) {
+    const thriftSpan = ThriftUtils.spanToThrift(span);
+    return new Promise((resolve, reject) => {
+      this.sender.append(thriftSpan, (numSpans: number, err?: string) => {
+        if (err) {
+          this.logger.error(`failed to add span: ${err}`);
+          reject(-1);
+        } else {
+          this.logger.debug('successful append for : %s', numSpans);
+          resolve(numSpans);
+        }
+      });
+    });
+  }
+
+  /**
+   * Publishes a list of root spans to Jaeger.
+   * @param rootSpans
+   */
+  publish(rootSpans: types.RootSpan[]) {
+    this.logger.debug('JeagerExport publishing');
+    for (const root of rootSpans) {
+      if (this.spanBuffer.indexOf(root) <= 0) {
+        this.onEndSpan(root);
+      }
+    }
+    return this.flush();
+  }
+
+  private flush() {
+    return new Promise((resolve, reject) => {
+      try {
```

Comment:
Is it possible for `this.sender.flush` to throw? If not I don't think the `try`/`catch` is necessary.

Reply:
> Is it possible for this.sender.flush to throw?

flush()  should not throw an exception, so we gonna keep the try catch, ok?